src/tools/wasm-link.c

```diff
@@ -759,61 +563,130 @@ static WasmBool write_combined_section(Context* ctx,
   return WASM_TRUE;
 }
 
+typedef struct ExportInfo {
+  Export* export;
+  InputBinary* binary;
+} ExportInfo;
+WASM_DEFINE_VECTOR(export_info, ExportInfo);
+
+static void resolve_symbols(Context* ctx) {
+  /* Create hashmap of all exported symbols from all inputs */
+  WasmBindingHash export_map;
+  WASM_ZERO_MEMORY(export_map);
+  ExportInfoVector export_list;
+  WASM_ZERO_MEMORY(export_list);
+
+  size_t i, j;
+  for (i = 0; i < ctx->inputs.size; i++) {
+    InputBinary* binary = &ctx->inputs.data[i];
+    for (j = 0; j < binary->exports.size; j++) {
+      Export* export = &binary->exports.data[j];
+      ExportInfo* info = wasm_append_export_info(ctx->allocator, &export_list);
+      info->export = export;
+      info->binary = binary;
+      WasmStringSlice name = wasm_dup_string_slice(ctx->allocator, export->name);
+      WasmBinding* binding = wasm_insert_binding(ctx->allocator, &export_map, &name);
```

Comment:
Might be a good idea to check for dupes, I don't think `wasm_insert_binding` does (though maybe it should)